File: CS/RichEditRibbonPerformanceBench/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.IO;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace RichEditRibbonPerformanceBench {
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm {

        Stopwatch sw = new Stopwatch();
        public Form1() {
            sw.Start();
            InitializeComponent();
        }

        protected override void OnShown(EventArgs e) {
            base.OnShown(e);
            sw.Stop();
            labelControl1.Text = "<b>Elapsed time:</b><br>" + sw.ElapsedMilliseconds.ToString() + " ms";
            File.AppendAllText(Directory.GetParent(Directory.GetCurrentDirectory()).Parent.FullName + "\\Results.txt", sw.ElapsedMilliseconds.ToString() + Environment.NewLine);
        }
    }
}

File: VB/RichEditRibbonPerformanceBench/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Diagnostics
Imports System.Drawing
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace RichEditRibbonPerformanceBench
    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Private sw As New Stopwatch()
        Public Sub New()
            sw.Start()
            InitializeComponent()
        End Sub

        Protected Overrides Sub OnShown(ByVal e As EventArgs)
            MyBase.OnShown(e)
            sw.Stop()
            labelControl1.Text = "<b>Elapsed time:</b><br>" & sw.ElapsedMilliseconds.ToString() & " ms"
            File.AppendAllText(Directory.GetParent(Directory.GetCurrentDirectory()).Parent.FullName & "\Results.txt", sw.ElapsedMilliseconds.ToString() & Environment.NewLine)
        End Sub
    End Class
End Namespace
